openmc/data/product.py

```diff
@@ -0,0 +1,187 @@
+        yield represents particles from prompt and delayed sources.
+    particle : str
+        What particle the reaction product is.
+    yield_ : openmc.data.Function1D
+        Yield of secondary particle in the reaction.
+
+    """
+
+    def __init__(self, particle='neutron'):
+        self.particle = particle
+        self.decay_rate = 0.0
+        self.emission_mode = 'prompt'
+        self.distribution = []
+        self.applicability = []
+        self.yield_ = Polynomial((1,))  # 0-order polynomial i.e. a constant
+
+    def __repr__(self):
+        if isinstance(self.yield_, Real):
+            return "<Product: {}, emission={}, yield={}>".format(
+                self.particle, self.emission_mode, self.yield_)
+        elif isinstance(self.yield_, Tabulated1D):
+            if np.all(self.yield_.y == self.yield_.y[0]):
+                return "<Product: {}, emission={}, yield={}>".format(
+                    self.particle, self.emission_mode, self.yield_.y[0])
+            else:
+                return "<Product: {}, emission={}, yield=tabulated>".format(
+                    self.particle, self.emission_mode)
+        else:
+            return "<Product: {}, emission={}, yield=polynomial>".format(
+                self.particle, self.emission_mode)
+
+    @property
+    def applicability(self):
+        return self._applicability
+
+    @property
+    def decay_rate(self):
+        return self._decay_rate
+
+    @property
+    def distribution(self):
+        return self._distribution
+
+    @property
+    def emission_mode(self):
+        return self._emission_mode
+
+    @property
+    def particle(self):
+        return self._particle
+
+    @property
+    def yield_(self):
+        return self._yield
+
+    @applicability.setter
+    def applicability(self, applicability):
+        cv.check_type('product distribution applicability', applicability,
+                      Iterable, Tabulated1D)
+        self._applicability = applicability
+
+    @decay_rate.setter
+    def decay_rate(self, decay_rate):
+        cv.check_type('product decay rate', decay_rate, Real)
+        cv.check_greater_than('product decay rate', decay_rate, 0.0, True)
+        self._decay_rate = decay_rate
+
+    @distribution.setter
+    def distribution(self, distribution):
+        cv.check_type('product angle-energy distribution', distribution,
+                      Iterable, AngleEnergy)
+        self._distribution = distribution
+
+    @emission_mode.setter
+    def emission_mode(self, emission_mode):
+        cv.check_value('product emission mode', emission_mode,
+                       ('prompt', 'delayed', 'total'))
+        self._emission_mode = emission_mode
+
+    @particle.setter
+    def particle(self, particle):
+        cv.check_type('product particle type', particle, str)
+        self._particle = particle
+
+    @yield_.setter
+    def yield_(self, yield_):
+        cv.check_type('product yield', yield_, Function1D)
```

Comment:
I suspect that I may be misunderstanding this, but the __repr__ method tests for a few different types of yield_. Does this setter accept a smaller subset of those types?